src/lib/frontend/typechecker.ml

```diff
@@ -2143,9 +2144,30 @@ let type_user_defined_type_body ~is_recursive env acc (loc, ls, s, body) =
     assert (not is_recursive); (* Abstract types are not recursive *)
     acc, env
 
-let rec type_decl (acc, env) d =
+let rec push_pop_env f n s =
+  if n <= 1 then
+    f s
+  else
+    let _ = f s in
```

Comment:
`let () = f s in`

Reply:
Can't use `()` because `f s` return an env when `f` is `Stack.pop`